### PR TITLE
docs: add contribution guidelines

### LINKED ISSUE #96: Repository would benefit from contribution guideline
Style guide(s), commit message conventions, etc.

### PR DESCRIPTION
**Pull request by x1ddos**
Should probably link to this from the readme but not sure where exactly yet.

Closes #96

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+8. Merge and submit the pull request. If you don't have merge permissions,
+a reviewer will do it for you.
+
+> **NOTE:** Before starting any coding work, please make sure to discuss the
+> issue first to agree on the optimal way of solving a particular problem.
+> If no issue exists, create one unless the change is trivial, such as
+> correcting typos and code code formatting.
+>
+> If you don't know where to start, look for issues labeled
+> with `good-first-issue`.
```
Comment (thisconnect) [inline CONTRIBUTING.md:27]: could link https://github.com/digitalbitbox/bitbox-wallet-app/issues?q=is%3Aissue+is%3Aopen+label%3Agood-first-issue

**Review by thisconnect — APPROVED** [review]
LGTM with a suggestion

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+## Branches
+
+When creating a new branch, give it a meaningful name. For example, `new-coin-xyz`.
```
Comment (benma) [inline CONTRIBUTING.md:31]: branches are local to developers and are pushed to their own forks, so it's up to them how to name it.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+Later, when a pull request is finally merged in a non fast-forward way, the merge
+commit would read:
+
+    Merge branch new-coin-xyz
```
Comment (benma) [inline CONTRIBUTING.md:35]: this happens only if the person merging checks out the branch under the same name. i usually check it out under the `pr/xyz` branch name or sth.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+larger than 500 lines, consider creating multiple pull requests or multiple
+commits depending on how closely the patches are related to each other.
+
+For example, if a change is much easier to do after a certain code reformatting,
+make the reformatting in a first commit and the actual change in the second.
+On the other hand, if a change includes multiple backend commits to create
+a new API endpoint as well as some UI modifications on the frontend, consider
+splitting backend and frontend into separate pull requests.
+
+A commit message consists of a subject, the first line, and a body separated
```
Comment (benma) [inline CONTRIBUTING.md:54]: maybe simpler to link to one of the many commit messages guides, e.g. the one from Go? can add anything on top below.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+To address review comments, amend an existing commit with:
+
+    git commit --amend
+
+and overwrite remote branch with:
+
+    git push --force
+
+> **NOTE:** If you're collaborating on a pull request with others, it is not
+> recommended to overwrite the remote branch and simple add more commits on top.
```
Comment (benma) [inline CONTRIBUTING.md:131]: this reas as if it is not recommended to add more commits, but you mean the opposite i think.

**Review by benma — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    This change modifies the _context_ to _summary_.
+
+> **NOTE:** That means the first line starts with lower case and does not end
+> with a period.
+
+For nontrivial changes, the subject alone is rarely self-explanatory.
+In this case, please add the body describing why this change is necessary,
+what it improves and how. Make sure the body lines fit in 72 characters,
+doesn't contain HTML, Markdown or any other markup language. Just plain text.
+
```
Comment (benma) [inline CONTRIBUTING.md:71]: maybe add: "The goal is that reading the commit message on any line using `git blame` explains why the line exists"

File: CONTRIBUTING.md (hunk 2 above)
Comment (x1ddos) [reply CONTRIBUTING.md:31]: in 99.9% cases local and remote branch take the same name. example of this pull request:

image

and `contrib-guidelines` is what will show up if you use standard commands provided by github.

File: CONTRIBUTING.md (hunk 3 above)
Comment (x1ddos) [reply CONTRIBUTING.md:35]: yeah. my point is when i see this highlighted merge commit:

image

it'll take me one extra roundtrip to github to understand what that commit actually merged. compare that to "merge branch fix-passphrase-last-step" - at least i know it's something related to the passphrase

Comment (x1ddos) [reply CONTRIBUTING.md:35]: > extra roundtrip to github

or `git log --graph ...something something`

File: CONTRIBUTING.md (hunk 5 above)
Comment (x1ddos) [reply CONTRIBUTING.md:131]: actually, i think it's too much detail here. i'll probably remove this note.

File: CONTRIBUTING.md (hunk 6 above)
Comment (x1ddos) [reply CONTRIBUTING.md:64]: added

File: CONTRIBUTING.md (hunk 5 above)
Comment (x1ddos) [reply CONTRIBUTING.md]: removed the note.

File: CONTRIBUTING.md (hunk 4 above)
Comment (x1ddos) [reply CONTRIBUTING.md:44]: i'm a bit hesitant to link to go or blog posts - those links can move away and i wouldn't want this guide to point to broken links. i think the guide covers enough of the basics.

i figured we can always update this if it turns out to not be enough. let's see.

**Comment by x1ddos** [comment]
ok, i link to contributing.md from the top readme

**Pull request merged**